src/koopmans/ml/_ml_models.py

```diff
@@ -67,51 +100,55 @@ def predict(self, x_test: np.ndarray) -> np.ndarray:
 
 
 class MLModel():
-    def __init__(self, type_of_ml_model='ridge_regression', is_trained: bool = False,
-                 X_train: Optional[np.ndarray] = None, Y_train: Optional[np.ndarray] = None):
+    def __init__(self, type_of_ml_model: str ='ridge_regression',
+                 X_train: Optional[np.ndarray] = None, Y_train: Optional[np.ndarray] = None, 
+                 save_dir: Optional[Path]=None):
         self.X_train = X_train
         self.Y_train = Y_train
         self.type_of_ml_model = type_of_ml_model
-        self.is_trained = is_trained
-        self.init_and_reset_model()
-
-    def init_and_reset_model(self):
-        if self.type_of_ml_model == 'ridge_regression':
-            self.model = RidgeRegressionModel()
-        elif self.type_of_ml_model == 'linear_regression':
-            self.model = LinearRegressionModel()
-        elif self.type_of_ml_model == 'mean':
-            self.model = MeanModel()
+        self.predictor: AbstractPredictor = self.init_and_reset_model(save_dir)
+
+    def init_and_reset_model(self, save_dir: Optional[Path]=None, sub_dir: Optional[str] = None) -> AbstractPredictor:
+        predictor: AbstractPredictor
+        predictor_classes: Dict[str, Type[AbstractPredictor]] = {'ridge_regression': RidgeRegressionModel, 'linear_regression': LinearRegressionModel, 'mean': MeanModel}
+        cls: Type[AbstractPredictor] = predictor_classes[self.type_of_ml_model]
+        if save_dir is None:
+            predictor = cls()
         else:
-            raise ValueError(f"{self.type_of_ml_model} is not implemented as a valid ML model.")
+            if sub_dir is None:
+                predictor = cls.load_from_file(save_dir)
+            else:
+                predictor = cls.load_from_file(save_dir / sub_dir)
+        return predictor
 
     def __repr__(self):
         if self.X_train and self.Y_train:
-            return f'{self.type_of_ml_model}(is_trained={self.model.is_trained}, ' \
+            return f'{self.type_of_ml_model}(is_trained={self.predictor.is_trained}, ' \
                    f'number_of_training_vectors={self.X_train.shape[0]}, ' \
                    f'input_vector_dimension={self.Y_train.shape[0]})'
         else:
-            return f'{self.type_of_ml_model}(is_trained={self.model.is_trained}, ' \
+            return f'{self.type_of_ml_model}(is_trained={self.predictor.is_trained}, ' \
                    'no training data has been added so far)'
 
     def todict(self):
-        dct = copy.deepcopy(dict(self.__dict__))
-        items_to_pop = ['model']
-        for item in items_to_pop:
-            dct.pop(item)
+        # Shallow copy
+        dct = dict(self.__dict__)
         return dct
```

Comment:
Do we need to add a `predictor` entry here to be consistent with `fromdict`?

Reply:
Always when an instance of this class is initialized with `__init__` the instance also contains a `predictor` entry. Do we still need to explicitly add a `predictor` entry?

Reply:
I have added `predictor` as an input variable for `__init__` then we don't have to pop it in `fromdict`